src/eos-crash-metrics.c

```diff
@@ -0,0 +1,194 @@
+  g_variant_dict_insert_value (&dict, "ostree_url", g_variant_new_string (ostree_url));
+
+  emtr_event_recorder_record_event_sync (emtr_event_recorder_get_default (),
+                                         PROGRAM_DUMPED_CORE_EVENT,
+                                         g_variant_dict_end (&dict));
+}
+
+static OstreeSysroot *
+load_ostree_sysroot (GError **error)
+{
+  OstreeSysroot *sysroot = ostree_sysroot_new_default ();
+  if (!ostree_sysroot_load (sysroot, NULL, error))
+    return NULL;
+
+  return sysroot;
+}
+
+static char *
+get_eos_ostree_repo_url (OstreeRepo *repo)
+{
+  GKeyFile *config = NULL;
+  g_autoptr(GError) error = NULL;
+  char *url = NULL;
+
+  config = ostree_repo_get_config (repo);
+
+  if (!(url = g_key_file_get_string (config, "remote \"eos\"", "url", &error)))
+    {
+      g_warning ("Unable to read OSTree config for eos remote URL: %s", error->message);
+      return NULL;
+    }
+
+  return url;
+}
+
+static char *
+get_eos_ostree_deployment_commit (OstreeSysroot *sysroot, OstreeRepo *repo)
+{
+  OstreeDeployment *deployment = ostree_sysroot_get_booted_deployment (sysroot);
+  GKeyFile *config = NULL;
+  g_autoptr(GError) error = NULL;
+  g_autofree char *refspec = NULL;
+  char *commit = NULL;
+
+  if (!deployment)
+    {
+      g_warning ("OSTree deployment is not currently booted, cannot read state");
+      return NULL;
+    }
+
+  config = ostree_deployment_get_origin (deployment);
+
+  if (!(refspec = g_key_file_get_string (config, "origin", "refspec", &error)))
+    {
+      g_warning ("Unable to read OSTree refspec for booted deployment: %s", error->message);
+      return NULL;
+    }
+
+  if (!ostree_repo_resolve_rev (repo, refspec, FALSE, &commit, &error))
+    {
+      g_warning ("Unable to resolve revision for refpsec %s: %s", refspec, error->message);
+      return NULL;
+    }
+
+  return commit;
+}
+
+static const char *blacklisted_prefixes[] =
+{
+  "/home",
+  "/sysroot/home"
+};
+
+static gboolean
+is_blacklisted_path (const char *path)
+{
+  const size_t n_blacklisted_prefixes = G_N_ELEMENTS (blacklisted_prefixes);
+  size_t i = 0;
+
+  for (; i < n_blacklisted_prefixes; ++i)
+    {
+      if (g_str_has_prefix (path, blacklisted_prefixes[i]))
+        return TRUE;
+    }
+
+  return FALSE;
+}
+
+/* The kernel gives us paths in the format !usr!bin!myprogram */
+static void
+normalize_path (char *path)
+{
+  size_t i = 0;
+
+  for (; i < strlen (path); ++i)
+    path[i] = path[i] == '!' ? '/' : path[i];
+}
+
+
+int
+main (int argc, char **argv)
+{
+  gchar *path = argv[1];
+  gint16 signal = 0;
+  gint64 timestamp = 0;
+  g_autoptr(OstreeSysroot) sysroot = NULL;
+  g_autoptr(OstreeRepo) repo = NULL;
+  g_autoptr(GError) error = NULL;
+  g_autofree char *url = NULL;
+  g_autofree char *commit = NULL;
+
+  if (argc != EXPECTED_NUMBER_ARGS + 1)
+    {
+      g_warning ("Not enough arguments passed (%i, expected %i).", argc - 1, EXPECTED_NUMBER_ARGS);
```

Comment:
Strictly speaking, this isn't really about "not enough arguments" being passed, but rather that the caller must pass three arguments